src/Umbraco.Cms.Api.Management/Services/Signs/HasPendingChangesSignProvider.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Umbraco.Cms.Api.Management.ViewModels;
+using Umbraco.Cms.Api.Management.ViewModels.Document;
+using Umbraco.Cms.Api.Management.ViewModels.Document.Collection;
+using Umbraco.Cms.Api.Management.ViewModels.Tree;
+using Umbraco.Cms.Core;
+
+
+namespace Umbraco.Cms.Api.Management.Services.Signs;
+
+public class HasPendingChangesSignProvider : ISignProvider
+{
+    private const string Alias = Constants.Conventions.Signs.Prefix + "PendingChanges";
+
+    public bool CanProvideSigns<TItem>()
+        where TItem : IHasSigns =>
+        typeof(TItem) == typeof(DocumentTreeItemResponseModel) ||
+        typeof(TItem) == typeof(DocumentCollectionResponseModel);
+
+    public Task PopulateSignsAsync<TItem>(IEnumerable<TItem> itemViewModels)
+        where TItem : IHasSigns
+    {
+        foreach (TItem item in itemViewModels)
+        {
+            switch (item)
```

Comment:
I tried a bit to see if we could find a way to use an interface here too to de-duplicate this code, but I couldn't get it to work unfortunately.  Mainly as there are two types we need to deal with the - the item and the variant.